### DISCUSSION TITLE
GeyserMC (and Floodgate) integration

### DISCUSSION THREAD

**Discussion by TheRobLP** [NamelessMC/Nameless#2140] [Ideas]
Relevant plugins:

https://github.com/GeyserMC/Geyser
https://github.com/GeyserMC/Floodgate

I would really love to see an integration with those plugins. Explanation as what those plugins do follows:

GeyserMC: GeyserMC is a "bridge" between Java and Bedrock allowing Bedrock players to join Java servers. It can be either run as independent client software, or server-side. Problem is: In order to join you need to have an Java account. Thats where Floodgate comes in.

Floodgate: Floodgate is a server-side GeyserMC addon solving the problem araising from GeyserMC. What it does is simple: Firstly, it adds a * behind every Bedrock username to prevent conflicts with Java. Secondly, it generates a unique UUID which would be impossible to get on Java. Those look like that: "00000000-0000-0000-0009-01f7ce49673f". I would really love to see GeyserMC added as an integration option into NamelessMC. Since Floodgate uses XBox-Auth to authenticate Bedrock players, you could use this for the integration too maybe. Thanks so much for your time reading trough this. I wish everyone of you a good day.

**Comment by samerton** [comment]
What exactly would you like to see in this integration?

We wouldn't be able to pull in a skin for example since it's not a valid Java UUID.

**Comment by TheRobLP** [comment]
First of all, I want to apologize for this late reply as I seriously forgot about this post. Secondly, I would love to see some sort of integration so Bedrock players can register. I think it could be done the same way that Floodgate allows Bedrock players to join the server, by authenticating them via XBox Live. To prevent that there is the same username on both Mojang and XBox, Bedrock players would need to use a special character before their name which would be customizable by the Admin, for example * . So my name on Pocket Edition would become *TheRobLP . Or what I would imagine would be easier to do is that registration for Bedrock players will only be possible ingame via registering trough the Nameless Plugin. Thanks in advance for any replies. 

Best Regards

**Comment by NoobcoreNetwork** [comment]
Updating this, floodgate 2 allows you to pull the bedrock skin so java players can see it, I'm not sure if theres a list somewhere online to check if its an actual bedrock account name but if you could do that and match the skin bedrock players could sign up for nameless also
